Source/WebCore/css/CSSRayValue.h

```diff
@@ -36,32 +36,38 @@ namespace WebCore {
 // https://drafts.fxtf.org/motion-1/#funcdef-offset-path-ray.
 class CSSRayValue final : public CSSValue {
 public:
-    static Ref<CSSRayValue> create(Ref<CSSPrimitiveValue>&& angle, CSSValueID size, bool isContaining)
+    static Ref<CSSRayValue> create(Ref<CSSPrimitiveValue>&& angle, CSSValueID size, bool isContaining, RefPtr<CSSValue>&& positionX, RefPtr<CSSValue>&& positionY)
     {
-        return adoptRef(*new CSSRayValue(WTFMove(angle), size, isContaining));
+        return adoptRef(*new CSSRayValue(WTFMove(angle), size, isContaining, WTFMove(positionX), WTFMove(positionY)));
     }
 
     String customCSSText() const;
 
     Ref<CSSPrimitiveValue> angle() const { return m_angle; }
     CSSValueID size() const { return m_size; }
     bool isContaining() const { return m_isContaining; }
+    RefPtr<CSSValue> positionX() const { return m_positionX; }
+    RefPtr<CSSValue> positionY() const { return m_positionY; }
 
     bool equals(const CSSRayValue&) const;
 
 private:
-    CSSRayValue(Ref<CSSPrimitiveValue>&& angle, CSSValueID size, bool isContaining)
+    CSSRayValue(Ref<CSSPrimitiveValue>&& angle, CSSValueID size, bool isContaining, RefPtr<CSSValue>&& positionX, RefPtr<CSSValue>&& positionY)
         : CSSValue(RayClass)
         , m_angle(WTFMove(angle))
         , m_size(size)
         , m_isContaining(isContaining)
+        , m_positionX(WTFMove(positionX))
+        , m_positionY(WTFMove(positionY))
     {
         ASSERT(m_angle->isAngle());
     }
 
     Ref<CSSPrimitiveValue> m_angle;
     CSSValueID m_size;
     bool m_isContaining;
+    RefPtr<CSSValue> m_positionX;
```

Comment:
Could these be a RefPtr to a CSSValuePair?

Then you get null-safety between the two of them (one of them can't be null when the other is non-null) and it's more-or-less a pair of Ref<CSSValue>. Plus you can just call `cssText` once when serializing and you get your space-separated pair of values.

Reply:
I could, although I was trying to be consistent with CSSCircleValue / CSSEllipseValue here, although arguably CSSValuePair could be used there too.